pkg/covermerger/covermerger.go

```diff
@@ -187,24 +187,55 @@ func mergeChanData(c *Config, recordsChan <-chan FileRecord) (map[string]*MergeR
 	targetFile := ""
 	var records []FileRecord
 	processedFiles := map[string]struct{}{}
+	var mu sync.Mutex
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+	var err error
+	var batchesRunning sync.WaitGroup
+	sem := semaphore.NewWeighted(c.Jobs)
```

Comment:
It might be somewhat cleaner if you explicitly create `c.Jobs` goroutines, each of which would do `for record := range recordsChan {`. When you close the channel, all those for loops will also stop.

Reply:
done